components/calendar/benches/datetime.rs

```diff
@@ -0,0 +1,224 @@
+                )
+                .unwrap();
+
+                // Conversion from ISO
+                let datetime_iso = DateTime::new_iso_datetime_from_integers(
+                    fx.year, fx.month, fx.day, fx.hour, fx.minute, fx.second,
+                )
+                .unwrap();
+                let mut converted_datetime_ethiopic =
+                    DateTime::new_from_iso(datetime_iso, Ethiopic);
+
+                run_calendar_benches(vec![
+                    &mut instantiated_datetime_ethiopic,
+                    &mut converted_datetime_ethiopic,
+                ]);
+            }
+        })
+    });
+
+    #[cfg(feature = "bench")]
+    group.bench_function("calendar/gregorian", |b| {
+        use icu::calendar::gregorian::Gregorian;
+
+        b.iter(|| {
+            for fx in &fxs.0 {
+                // Instantion from int. Version with nanoseconds, so supplying nanosecond
+                let mut instantiated_datetime_gregorian =
+                    DateTime::new_gregorian_datetime_from_integers(
+                        fx.year, fx.month, fx.day, fx.hour, fx.minute, fx.second, 0
+                    )
+                    .unwrap();
+
+                // Conversion from ISO
+                let datetime_iso = DateTime::new_iso_datetime_from_integers(
+                    fx.year, fx.month, fx.day, fx.hour, fx.minute, fx.second
+                )
+                .unwrap();
+                let mut converted_datetime_gregorian =
+                    DateTime::new_from_iso(datetime_iso, Gregorian);
+
+                run_calendar_benches(vec![
+                    &mut instantiated_datetime_gregorian,
+                    &mut converted_datetime_gregorian,
+                ]);
+            }
+        })
+    });
+
+    #[cfg(feature = "bench")]
+    group.bench_function("calendar/indian", |b| {
+        use icu::calendar::indian::Indian;
+
+        b.iter(|| {
+            for fx in &fxs.0 {
+                // Instantion from int
+                let mut instantiated_datetime_indian = DateTime::new_indian_datetime(
+                    fx.year, fx.month, fx.day, fx.hour, fx.minute, fx.second,
+                )
+                .unwrap();
+
+                // Conversion from ISO
+                let datetime_iso = DateTime::new_iso_datetime_from_integers(
+                    fx.year, fx.month, fx.day, fx.hour, fx.minute, fx.second,
+                )
+                .unwrap();
+                let mut converted_datetime_indian = DateTime::new_from_iso(datetime_iso, Indian);
+
+                run_calendar_benches(vec![
+                    &mut instantiated_datetime_indian,
+                    &mut converted_datetime_indian,
+                ]);
+            }
+        })
+    });
+
+    #[cfg(feature = "bench")]
+    group.bench_function("calendar/julian", |b| {
+        use icu::calendar::julian::Julian;
+
+        b.iter(|| {
+            for fx in &fxs.0 {
```

Comment:
issue: So I was hoping most of this could be done generically too, ideally the code just needs one or two lines per calendar.

You'd need to write a function that looks something like

```rust
fn bench_calendar<C: Calendar>(group: .., name: &str, fxs: ..., calendar_init: impl Fn(i32, i32, i32, i32, i32, i32) -> DateTime<C>) {...}

bench_calendar(&group, "calendar/julian", &fxs, |y, m, d, h, min, s| DateTime::new_julian_datetime(y, m, d, h, min, s).unwrap());
bench_calendar(&group, "calendar/buddhist", &fxs, |y, m, d, h, min, s| DateTime::new_buddhist_datetime(y, m, d, h, min, s).unwrap());
```

(the `Fn` will likely have different integer types as bounds but i can't remember them)

Reply:
Ohhh, this makes tons more sense, thank you for providing the skeleton! For my understanding, what is the closure accomplishing for the passed value of `|y, m, d, h, min, s| DateTime::new_julian_datetime(y, m, d, h, min, s).unwrap()`? Why can't we instead call `DateTime::new_julian_datetime`, and unwrap within the function?